R/model_options.R

```diff
@@ -0,0 +1,76 @@
+#' Get model options declarative UI from naomi and input data
+#'
+#' Get's UI template from Naomi and substitutes any params within the
+#' template with real values.
+#'
+#' @param options_template Naomi template
+#' @param shape Path to shape file
+#' @param survey Path to survey file
+#' @param programme Path to optional programme file
+#' @param anc Path to optional ANC file
+#'
+#' @return The model options declarative JSON UI.
+#' @keywords internal
+do_endpoint_model_options <- function(options_template, shape, survey,
+                                      programme, anc) {
+  regions <- read_geojson_regions(shape)
+  parent_region <- regions[!grepl("\\.", regions)]
+  if (length(parent_region) != 1) {
+    stop(sprintf("Should have located one parent regions but found regions %s.",
```

Comment:
if this is a likely problem, should you be checking that all regions have the same start? (`length(unique(sub("\\..+$", "", regions))) == 1`)